lighthouse-core/test/gather/gatherers/image-elements-test.js

```diff
@@ -0,0 +1,282 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+const ImageElements = require('../../../gather/gatherers/image-elements.js');
+const NetworkRecorder = require('../../../lib/network-recorder.js');
+const {createMockContext} = require('../../fraggle-rock/gather/mock-driver.js');
+const {makeParamsOptional} = require('../../test-utils.js');
+
+const devtoolsLog = /** @type {LH.DevtoolsLog} */ (require('../../fixtures/traces/lcp-m78.devtools.log.json')); // eslint-disable-line max-len
+const networkRecords = NetworkRecorder.recordsFromLogs(devtoolsLog);
+
+/** @type {LH.Artifacts.ImageElement} */
+const imageElement = {
+  src: 'https://www.paulirish.com/avatar150.jpg',
+  srcset: '',
+  displayedWidth: 200,
+  displayedHeight: 200,
+  clientRect: {
+    top: 50,
+    bottom: 250,
+    left: 50,
+    right: 250,
+  },
+  attributeWidth: '',
+  attributeHeight: '',
+  cssWidth: undefined,
+  cssHeight: undefined,
+  _privateCssSizing: undefined,
+  cssComputedPosition: 'absolute',
+  isCss: false,
+  isPicture: false,
+  isInShadowDOM: false,
+  cssComputedObjectFit: '',
+  cssComputedImageRendering: '',
+  node: {
+    lhId: '__nodeid__',
+    devtoolsNodePath: '1,HTML,1,BODY,1,DIV,1,IMG',
+    selector: 'body > img',
+    nodeLabel: 'img',
+    snippet: '<img src="https://www.paulirish.com/avatar150.jpg">',
+    boundingRect: {
+      top: 50,
+      bottom: 250,
+      left: 50,
+      right: 250,
+      width: 200,
+      height: 200,
+    },
+  },
+};
+
+jest.useFakeTimers();
+
+function mockImageElements() {
+  const gatherer = new ImageElements();
+  return {
+    ...gatherer,
+    _getArtifact: jest.fn(makeParamsOptional(gatherer._getArtifact)),
```

Comment:
`makeParamsOptional` was super helpful in gather-runner for a smoother transition, but thusfar I think we've been able to avoid it in most of the greenfield tests by creating good mock creation utils and things like `network-records-to-devtools-log.js`.

Is the delta here very large or do you think it's achievable to make do without `makeParamsOptional`?

Reply:
It's probably possible with a mock creation util for elements and network requests in this file. I will make the attempt and report back if there are any other major issues.

Reply:
Indeed it is possible.